src/keylessBackup/KeylessBackupProgress.tsx

```diff
@@ -24,21 +29,77 @@ const TAG = 'keylessBackup/KeylessBackupProgress'
 function KeylessBackupProgress({
   route,
 }: NativeStackScreenProps<StackParamList, Screens.KeylessBackupProgress>) {
-  const keylessBackupStatus = useSelector(keylessBackupStatusSelector)
-
-  const { t } = useTranslation()
-
   // Disable back button on Android
   useEffect(() => {
     const backPressListener = () => true
     BackHandler.addEventListener('hardwareBackPress', backPressListener)
     return () => BackHandler.removeEventListener('hardwareBackPress', backPressListener)
   }, [])
 
-  // TODO(ACT-781): Implement Restore flow designs
   if (route.params.keylessBackupFlow === KeylessBackupFlow.Restore) {
-    return <></>
+    return <Restore />
+  } else {
+    return <Setup />
   }
+}
+
+function Restore() {
+  const keylessBackupStatus = useSelector(keylessBackupStatusSelector)
+  const localCurrencyCode = useLocalCurrencyCode()
+
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  switch (keylessBackupStatus) {
+    case KeylessBackupStatus.InProgress: {
+      return InProgress(t('keylessBackupStatus.restore.inProgress.title'))
```

Comment:
nit: maybe make this a react component instead of a function? or use a lower case fn name? e.g., renderInProgress